flink-runtime/src/main/java/org/apache/flink/runtime/heartbeat/HeartbeatServices.java

```diff
@@ -100,37 +62,31 @@ public <I, O> HeartbeatManager<I, O> createHeartbeatManager(
      * @param <O> Type of the outgoing payload
      * @return A new HeartbeatManager instance which actively sends heartbeats
      */
-    public <I, O> HeartbeatManager<I, O> createHeartbeatManagerSender(
+    <I, O> HeartbeatManager<I, O> createHeartbeatManagerSender(
             ResourceID resourceId,
             HeartbeatListener<I, O> heartbeatListener,
             ScheduledExecutor mainThreadExecutor,
-            Logger log) {
-
-        return new HeartbeatManagerSenderImpl<>(
-                heartbeatInterval,
-                heartbeatTimeout,
-                failedRpcRequestsUntilUnreachable,
-                resourceId,
-                heartbeatListener,
-                mainThreadExecutor,
-                log);
-    }
+            Logger log);
 
     /**
      * Creates an HeartbeatServices instance from a {@link Configuration}.
      *
      * @param configuration Configuration to be used for the HeartbeatServices creation
      * @return An HeartbeatServices instance created from the given configuration
      */
-    public static HeartbeatServices fromConfiguration(Configuration configuration) {
+    static HeartbeatServices fromConfiguration(Configuration configuration) {
         long heartbeatInterval = configuration.getLong(HeartbeatManagerOptions.HEARTBEAT_INTERVAL);
 
         long heartbeatTimeout = configuration.getLong(HeartbeatManagerOptions.HEARTBEAT_TIMEOUT);
 
         int failedRpcRequestsUntilUnreachable =
                 configuration.get(HeartbeatManagerOptions.HEARTBEAT_RPC_FAILURE_THRESHOLD);
 
-        return new HeartbeatServices(
+        return new HeartbeatServicesImpl(
                 heartbeatInterval, heartbeatTimeout, failedRpcRequestsUntilUnreachable);
     }
+
+    static HeartbeatServices noOp() {
```

Comment:
That's kind of related to my previous comment: We could remove that if we decide to move the `NoOpheartbeatServices` into the test code, I guess.